src/frontend/screens/Library/components/InstallModal/index.tsx

```diff
@@ -153,6 +158,36 @@ export default React.memo(function InstallModal({
       setPlatformToInstall(availablePlatforms[0].value as InstallPlatform)
   }, [availablePlatforms])
 
+  const channelRequiresAccessCode =
+    !!selectedChannel?.license_config.access_codes
+
+  useEffect(() => {
```

Comment:
[I like to avoid useEffect's whenever it is possible](https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers). I see an opportunity here to use a form and display the error on submit. 

It'd be something like this:

```tsx
function handleSubmit(event: FormEvent) {
 if (needsAccessCode) {
   const valid = await window.api.validate()
    if (!valid) {
       setErrorText('invalid code')
       return
     }
  }
}
```
What do you think?

Reply:
We'd also show the install button always, it'd be always enabled and would trigger the form submit.

Reply:
hmm interesting. let me check with the designers

Reply:
Based on the conversation with the designer the only change remaining would be to make the install button always visible. I believe it should only be enabled when the code is valid.

Reply:
sounds good. I did this and also added the red error stroke and the green success text